recipes/full_finetune_distributed.py

```diff
@@ -846,6 +855,12 @@ def train(self) -> None:
                     if self._lr_scheduler is not None:
                         self._lr_scheduler.step()
 
+                    # If float8 training is enabled, perform a single all-reduce to compute the
+                    # scale for all float8 parameters efficiently instead of doing many small
+                    # all-reduces for each parameter
+                    if self._enable_fp8_training and self.dp_degree > 1:
+                        precompute_float8_dynamic_scale_for_fsdp(self._model)
```

Comment:
I think this should only be enabled for tensorwise scaling, the other recipes do not support float8 all-gather for FSDP right now.